authentication-client/authentication-client-java/src/main/java/co/cask/cdap/security/authentication/client/basic/BasicAuthenticationClient.java

```
@@ -0,0 +1,183 @@
+    }
+
+    if (accessToken == null) {
+      accessToken = fetchAccessToken();
+    }
+    return accessToken;
+  }
+
+  @Override
+  public void invalidateToken() {
+    accessToken = null;
+  }
+
+  @Override
+  public boolean isAuthEnabled() throws IOException {
+    if (authEnabled == null) {
+      String strAuthUrl = getAuthURL();
+      authEnabled = StringUtils.isNotEmpty(strAuthUrl);
+      if (authEnabled) {
+        authUrl = URI.create(strAuthUrl);
+      }
+    }
+    return authEnabled;
+  }
+
+  private AccessToken fetchAccessToken() throws IOException {
+    if (StringUtils.isEmpty(username) || StringUtils.isEmpty(password)) {
+      throw new IllegalStateException("Base authentication client is not configured!");
+    }
+
+    LOG.debug("Authentication is enabled in the gateway server. Authentication URI {}.", authUrl);
+    HttpGet getRequest = new HttpGet(authUrl);
+
+    String auth = Base64.encodeBase64String(String.format("%s:%s", username, password).getBytes());
+    auth = auth.replaceAll("(\r|\n)", StringUtils.EMPTY);
+    getRequest.addHeader(HttpHeaders.AUTHORIZATION, AUTHENTICATION_HEADER_PREFIX_BASIC + auth);
+
+    HttpResponse httpResponse = httpClient.execute(getRequest);
+    RestClientUtils.verifyResponseCode(httpResponse);
+
+    JsonElement jsonElement = RestClientUtils.toJsonElement(httpResponse.getEntity());
+    Map<String, String> responseMap = GSON.fromJson(jsonElement, new TypeToken<Map<String, String>>() {
+    }.getType());
+    String tokenValue = responseMap.get(ACCESS_TOKEN_KEY);
+    String tokenType = responseMap.get(TOKEN_TYPE_KEY);
+    if (StringUtils.isEmpty(tokenValue) || StringUtils.isEmpty(tokenType)) {
+      throw new IOException("Unexpected response was received from the authentication server.");
+    }
+
+    String expiresInStr = responseMap.get(EXPIRES_IN_KEY);
+    Long expiresIn = StringUtils.isNotEmpty(expiresInStr) ? Long.valueOf(expiresInStr) : Long.valueOf(0);
```

Since the documentation says expires_in will be returned in successful responses, we should just throw an exception when it is not present.